CMakePresets.json

```diff
@@ -1,141 +1,174 @@
+            },
+            "environment": {
+                "LADYBIRD_SOURCE_DIR": "${fileDir}",
+                "VCPKG_ROOT": "${fileDir}/Toolchain/Tarballs/vcpkg",
+                "VCPKG_BINARY_SOURCES": "clear;files,${fileDir}/Toolchain/Build/vcpkg-binary-cache,readwrite"
+            },
+            "vendor": {
+                "jetbrains.com/clion": {
+                    "toolchain": "Default"
+                }
+            }
+        },
+        {
+            "name": "Debug",
+            "inherits": "default",
+            "displayName": "Debug Config",
+            "description": "Debug build using Ninja generator",
+            "binaryDir": "${fileDir}/Build/ladybird-debug",
+            "cacheVariables": {
+                "CMAKE_BUILD_TYPE": "Debug",
+                "VCPKG_OVERLAY_TRIPLETS": "${fileDir}/Meta/CMake/vcpkg/debug-triplets"
+            }
+        },
+        {
+            "name": "Sanitizer",
+            "inherits": "Debug",
+            "displayName": "Sanitizer Config",
+            "description": "Debug build using Sanitizers",
+            "binaryDir": "${fileDir}/Build/ladybird-sanitizers",
+            "cacheVariables": {
+                "ENABLE_UNDEFINED_SANITIZER": "ON",
+                "ENABLE_ADDRESS_SANITIZER": "ON",
+                "VCPKG_OVERLAY_TRIPLETS": "${fileDir}/Meta/CMake/vcpkg/sanitizer-triplets"
+            }
+        },
+        {
+            "hidden": true,
+            "name": "vcpkg_ci",
+            "description": "Use the GitHub Actions vcpkg cache",
+            "environment": {
+                "VCPKG_BINARY_SOURCES": "clear;x-gha,readwrite"
+            }
+        },
+        {
+            "name": "CI",
+            "inherits": [
+                "vcpkg_ci",
+                "default"
+            ],
+            "displayName": "Non-Sanitizer CI Config"
+        },
+        {
+            "name": "Sanitizer_CI",
+            "inherits": [
+                "vcpkg_ci",
+                "Sanitizer"
+            ],
+            "displayName": "Sanitizer CI Config",
+            "cacheVariables": {
+                "CMAKE_BUILD_TYPE": ""
+            }
+        },
+        {
+            "name": "Fuzzers_CI",
+            "inherits": [
+                "vcpkg_ci",
+                "default"
+            ],
+            "displayName": "Fuzzers Config",
+            "description": "Fuzzers build with GitHub Actions cache",
+            "binaryDir": "${fileDir}/Build/ladybird-fuzzers",
+            "cacheVariables": {
+                "CMAKE_BUILD_TYPE": "",
+                "ENABLE_QT": "OFF",
+                "ENABLE_FUZZERS_LIBFUZZER": "ON",
+                "ENABLE_ADDRESS_SANITIZER": "ON"
+            }
+        },
+        {
+            "name": "VisualStudio",
+            "displayName": "Visual Studio 2022 Config",
+            "description": "Build using Visual Studio 2022 generator",
+            "generator": "Visual Studio 17 2022",
```

Comment:
Using MSVC as a compiler is a non-goal. The only native compiler that will work on windows without serious rework of core libraries is `clang-cl.exe` (or `clang.exe`).

Reply:
As a consequence, the warning changes aren't needed: clang accepts the same warning flags as clang-cl.

Reply:
When clang-cl is used in CMake build, it does not define MSVC variable. So these warning changes needed to be guarded with `NOT (MSVC)` case

Reply:
when it comes to presets i've never worked with conditionals so have no clue on how I can make the preset generic?
could I get some help with this

Reply:
Looking around, this will require a bit more elbow grease. So, we can skip the conditional preset idea for now. <https://discourse.cmake.org/t/presets-conditional-config/3864/4>

fwiw, `clang-cl.exe` *does* cause `if (MSVC)` to trigger. `clang.exe` does. this is related to CMAKE_CXX_COMPILER_FRONTEND_VARIANT.